### DISCUSSION TITLE
How to build and deploy to sub-folder

### DISCUSSION THREAD

**Discussion by HouseSitSearch** [quasarframework/quasar#14984] [General - Components / Directives / etc] [answered]
I've just installed the latest version of quasar, then run the "yarn create quasar" and used the suggested options to create a basic vite setup.
After the create process has finished, I've simply run the command
quasar build

This has created a dist folder. I've then used IIS on Windows, to point that dist folder to a new Virtual Directory, and navigated to the "spa" subfolder that is created in my browser, i.e.
http://localhost/testapp/dist/spa

The index.html file is found and loaded, but this then makes calls to load additional files such as:
http://localhost/assets/index.4c5d30fa.js
http://localhost/assets/index.b76c8355.css
http://localhost/icons/favicon-128x128.png
http://localhost/icons/favicon-32x32.png

How can this be made to work as expected, as I'd like to develop something that can run in a sub-folder in a production environment?

Thanks in advance for your help,

Mark

**Comment by suleimanas** [comment, marked as answer]
In quasar.config > build configuration options:```
```

...
build: {
  // Public path of your app. default '/'
  publicPath: "/testapp/",

  // Build dist
  distDir: "./anyFolder"
}
...
```

https://quasar.dev/quasar-cli-vite/quasar-config-js#build


relative publicPath with vite
```
    build: {
      publicPath: "",
      extendViteConf(viteConf, { isServer, isClient }) {
        viteConf.base = "";
      },
      ...
    }
```
ref: https://github.com/quasarframework/quasar/issues/8513#issuecomment-1127654470


with webpack https://webpack.js.org/guides/public-path/#automatic-publicpath

**Reply by HouseSitSearch** [reply]
Thanks for your help, it's really appreciated.

**Comment by HouseSitSearch** [comment]
Thanks for the helpful reply. 

Does this mean that it's not possible to use purely relative paths, so that the dist/spa output can be dropped into any folder and simply work?

I'm thinking of how this could be integrated into wordpress, so that it could run in the root folder, a category sub-folder, or a post etc.

**Reply by suleimanas** [reply]
Yes it is possible to have relative publicPath
```
    build: {
      publicPath: "",
      extendViteConf(viteConf, { isServer, isClient }) {
        viteConf.base = "";
      },
      ...
    }
```
ref: https://github.com/quasarframework/quasar/issues/8513#issuecomment-1127654470

**Reply by HouseSitSearch** [reply]
You little beauty! That's genius and works perfectly.

I also have some old project that use webpack, so is this possible in webpack setups too?

**Reply by suleimanas** [reply]
sure same idea https://webpack.js.org/guides/public-path/#automatic-publicpath

Note: don't mark your comment as the answer, come on :)

**Reply by HouseSitSearch** [reply]
Thanks for the webpack tip, I'll give that a go next.

Sorry, but there's no "Mark as answer" button for your sub-comment. I wanted to mark your "extendViteConf" comment as the answer, but the button isn't there.

Your other answer isn't what I, or possibly other people, are looking for, but that 2nd answer is perfect. So, which of the 2 "Mark as answer" buttons should I click? I've only got 2 options, 1 for your 1st comment, which isn't what I wanted, and 1 for my subsequent question, which contains the right answer in the sub messages.

**Reply by suleimanas** [reply]
glad to help, I've edited the first answer
